bindings/nodejs/lib/bindings.ts

```diff
@@ -1,18 +1,29 @@
 // Copyright 2021 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
+import type { EventType } from '../types';
 import type { MessageHandler } from './MessageHandler';
+
 // @ts-ignore: path is set to match runtime transpiled js path
 import addon = require('../../build/Release/index.node');
 
 const {
     initLogger,
     sendMessage,
     messageHandlerNew,
-    listen,
+    listenRust,
     destroy,
 } = addon;
 
+const listen = (
+    eventTypes: EventType[],
+    callback: (error: Error, result: string) => void,
+    handler: MessageHandler,
+): Promise<void> => {
+    listenRust(eventTypes, callback, handler);
+    return Promise.resolve();
```

Comment:
this is needed so that the signature is the same when copying to wasm. However wasm cant just spawn a new thread to block, which is why i chose this method